liquidity-pool-changes/src/agent.js

```diff
@@ -0,0 +1,194 @@
+const BigNumber = require('bignumber.js');
+const axios = require('axios');
+
+const {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} = require('forta-agent');
+
+// load agent configurations
+const config = require('../agent-config.json');
+
+const utils = require('./utils');
+
+// used to store initialization data once you pass it into the provideHandleBlock() function
+const initializeData = {};
+
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    // setup ethers.js provider to interact with contracts
+    data.provider = getEthersProvider();
+
+    // get abi
+    data.erc20Abi = utils.getAbi('ERC20.json');
+
+    // get liquidityThresholdPercentChange
+    data.liquidityThresholdPercentChange = new BigNumber(
+      config.liquidityThresholdPercentChange,
+    );
+
+    // initalize previousLiquidity to undefined
+    data.previousLiquidity = undefined;
+    /* eslint-disable no-param-reassign */
+  };
+}
+
+async function getTokenPrices(token0Address, token1Address) {
+  const coingeckoApiUrl = 'https://api.coingecko.com/api/v3/simple/token_price/ethereum?';
+  const addressQuery = `contract_addresses=${token0Address},${token1Address}`;
+  const vsCurrency = '&vs_currencies=usd';
+
+  const url = coingeckoApiUrl + addressQuery + vsCurrency;
+  const { data } = await axios.get(url);
+
+  // parse response to convert to BigNumber.js
+  const usdPerToken0 = new BigNumber(data[token0Address.toLowerCase()].usd);
+  const usdPerToken1 = new BigNumber(data[token1Address.toLowerCase()].usd);
+
+  return { token0Price: usdPerToken0, token1Price: usdPerToken1 };
+}
+
+// amountBN is the amount of tokens in the liquidity pool in BigNumber.js format
+async function getTokenUSDValue(amountBN, tokenPrice, tokenContract) {
+  // get the decimal scaling for this token
+  let decimals;
+  try {
+    decimals = await tokenContract.decimals();
+  } catch {
+    return undefined;
+  }
+  decimals = new BigNumber(decimals); // convert from ether.js bignumber to bignumber.js
```

Comment:
Converting directly from one implementation of BigNumber to another is undefined behavior, convert the ethers.js bignumber to a string first:
```suggestion
  decimals = new BigNumber(decimals.toString()); // convert from ether.js bignumber to bignumber.js
```